vuestorefront/schemas/product.py

```diff
@@ -107,22 +107,27 @@ class ProductQuery(graphene.ObjectType):
     @staticmethod
     def resolve_product(self, info, id=None, slug=None, barcode=None):
         env = info.context["env"]
-        Product = product = env["product.template"].sudo()
+        ProductTemplate = product = env["product.template"].sudo()
         domain = None
-        base_domain = [('is_published', '=', True)]
+        base_domain = []
         if id:
             domain = base_domain + [('id', '=', id)]
         elif slug:
             domain = base_domain + [('website_slug', '=', slug)]
         elif barcode:
             domain = base_domain + [('barcode', '=', barcode)]
         if domain is not None:
-            product = Product.search(domain, limit=1)
+            product = ProductTemplate.search(domain, limit=1)
+        if (
+            ProductTemplate.is_vsf_published_only(product=product)
```

Comment:
When accessing single product filters will not be used.
Would be needed to check if the product is part user's partner pricelist.